src/SurfaceFluxes.jl

```diff
@@ -357,50 +363,76 @@ function obukhov_length(
+
+    thermo_params = SFP.thermodynamics_params(param_set)
+    grav = SFP.grav(param_set)
+    cp_d = SFP.cp_d(param_set)
+    DSEᵥ_sfc = TD.virtual_dry_static_energy(thermo_params, ts_sfc(sc), grav * z_sfc(sc))
+    DSEᵥ_in = TD.virtual_dry_static_energy(thermo_params, ts_in(sc), grav * z_in(sc))
+    ΔDSEᵥ = DSEᵥ_in - DSEᵥ_sfc
+    tol_neutral = FT(cp_d / 10)
+
+    function solve_lmo(ΔDSEᵥ)
+        if abs(ΔDSEᵥ) <= tol_neutral
+            L_MO = z_in(sc) / tol_neutral
+            sol = nothing
+            is_converged = true
+        else
+            sol = RS.find_zero(root_l_mo, RS.NewtonsMethodAD(sc.L_MO_init), soltype, tol, maxiter)
+            L_MO = sol.root
+            is_converged = sol.converged
+        end
+        return sol, L_MO, is_converged
+    end
+    sol, L_MO, is_converged = solve_lmo(ΔDSEᵥ)
+    if sol != nothing && !sol.converged
+        KA.@print("-----------------------------------------\n")
+        KA.@print("maxiter reached in SurfaceFluxes.jl:\n")
+        KA.@print(", T_in = ", TD.air_temperature(thermo_params, ts_in(sc)))
+        KA.@print(", T_sfc = ", TD.air_temperature(thermo_params, ts_sfc(sc)))
+        KA.@print(", q_in = ", TD.total_specific_humidity(thermo_params, ts_in(sc)))
+        KA.@print(", q_sfc = ", TD.total_specific_humidity(thermo_params, ts_sfc(sc)))
+        KA.@print(", u_in = ", u_in(sc))
+        KA.@print(", u_sfc = ", u_sfc(sc))
+        KA.@print(", z0_m = ", z0(sc, UF.MomentumTransport()))
+        KA.@print(", z0_b = ", z0(sc, UF.HeatTransport()))
+        KA.@print(", Δz = ", Δz(sc))
+        KA.@print(", ΔDSE = ", ΔDSE)
         if error_on_non_convergence()
-            KA.@print("-----------------------------------------\n")
-            KA.@print("maxiter reached in SurfaceFluxes.jl:\n")
-            KA.@print(", T_in = ", TD.air_temperature(thermo_params, ts_in(sc)))
-            KA.@print(", T_sfc = ", TD.air_temperature(thermo_params, ts_sfc(sc)))
-            KA.@print(", q_in = ", TD.total_specific_humidity(thermo_params, ts_in(sc)))
-            KA.@print(", q_sfc = ", TD.total_specific_humidity(thermo_params, ts_sfc(sc)))
-            KA.@print(", u_in = ", u_in(sc))
-            KA.@print(", u_sfc = ", u_sfc(sc))
-            KA.@print(", z0_m = ", z0(sc, UF.MomentumTransport()))
-            KA.@print(", z0_b = ", z0(sc, UF.HeatTransport()))
-            KA.@print(", Δz = ", Δz(sc))
             error("Unconverged Surface Fluxes.")
         else
-            # KA.@print("Warning: Unconverged Surface Fluxes. Returning neutral condition solution \n")
-            L_MO = sol.root
+            KA.@print("Warning: Unconverged Surface Fluxes. Returning iteration history.")
+            if soltype isa CompactSolution
+                @show sol.root
+            else
+                @show sol.err_history
+                @show sol.root_history
+                KA.@print("-----------------------------------------\n")
+            end
```

Comment:
This needs to be reverted. Using `@show` inside a broadcast is a GPU issue.

Reply:
Ah, thanks! So we just cannot show this information?

Reply:
`@show` doesn't work on the GPU, so no, not like this. That's why we using `KA.@print` Also, for practicality purposes, it should be conditional. Doing this can result in crashed simulations with extremely bloated output (causing memory issues).

Reply:
Ok thanks. I'll open a PR to fix this.